Rebus.ServiceProvider.Tests/PolymorphicMessageHandlerActivation.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+using Rebus.Activation;
+using Rebus.Handlers;
+using Rebus.Retry.Simple;
+using Rebus.Routing.TypeBased;
+using Rebus.Transport;
+using Rebus.Transport.InMem;
+
+namespace Rebus.ServiceProvider.Tests
+{
+    /// <summary>
+    ///     Testing the polymorphic message handler activation.
+    /// </summary>
+    [TestFixture]
+    public class PolymorphicMessageHandlerActivation
```

Comment:
Suggestion: Inherit from `FixtureBase`, as this will provide a nifty `Using` method that can keep a stack of `IDisposable`s behind the scenes.....